mindmaps-engine/src/main/java/io/mindmaps/util/ConfigProperties.java

```diff
@@ -52,16 +58,39 @@
     public static final String CURRENT_DIR_SYSTEM_PROPERTY = "mindmaps.dir";
     public static final String CONFIG_FILE_SYSTEM_PROPERTY = "mindmaps.conf";
 
+    private final Logger LOG = LoggerFactory.getLogger(ConfigProperties.class);
 
+    private final int MAX_NUMBER_OF_THREADS = 120;
     private Properties prop;
     private static ConfigProperties instance = null;
     private String configFilePath = null;
+    private int numOfThreads = -1;
```

Comment:
worst case: one thread will see it as -1 even if another one is modifying it and will force ConfigProperties to recompute the number of threads, which should always be the same. 
But I can make it atomic if it makes more sense